corehq/reports.py

```diff
@@ -166,7 +168,23 @@ def _get_configurable_reports(project):
     """
     User configurable reports
     """
-    configs = ReportConfiguration.by_domain(project.name) + StaticReportConfiguration.by_domain(project.name)
+    try:
+        configs = ReportConfiguration.by_domain(project.name)
+    except BadSpecError as e:
+        logging.exception(e)
+
+        # Pick out the UCRs that don't have spec errors
+        configs = []
+        for config_id in get_doc_ids_in_domain_by_class(project.name, ReportConfiguration):
+            try:
+                configs.append(ReportConfiguration.get(config_id))
+            except BadSpecError as e:
+                logging.error("%s with report config %s" % (e.message, config_id))
+    try:
+        configs.extend(StaticReportConfiguration.by_domain(project.name))
```

Comment:
Do we care about getting the statically-defined reports that don't have spec errors, like we do for the UCRs above? If so, we could simplify this a little into one outer `try`, and chain ReportConfiguration and StaticReportConfiguration in the `for` loop.